zilliqa/src/consensus.rs

```diff
@@ -526,6 +527,28 @@ impl Consensus {
             return Ok(None);
         }
 
+        let now = SystemTime::now();
+        if !during_sync
+            && block.timestamp() + self.config.consensus.maximum_delay_for_received_block < now
+        {
+            let diff = now.duration_since(SystemTime::UNIX_EPOCH)?
+                - (block.timestamp().duration_since(SystemTime::UNIX_EPOCH)?
+                    + self.config.consensus.maximum_delay_for_received_block);
+            warn!(
+                "Rejecting block - It's too old, diff is {}ms",
```

Comment:
Why do we care about this? If old blocks trog in slowly over a period of hours, why do we care?

Reply:
Also, why do we have two settings for this? check_block() checks `config.allowed_timestamp_skew` .

Reply:
I'd suggest for now we only check the timestamp is increasing as per the [yellow paper](https://ethereum.github.io/yellowpaper/paper.pdf). It's an imperfect solutions, but we can improve it in a separate issue later on...